### PR TITLE
HS-1802: Welcome Guide Updates

### PR DESCRIPTION
**Pull request by scthompsonopen**
## Description
Based on Emails, tickets and DMs received, we've received requests to make the following adjustments (so far, this PR is evolving) to the welcome guide.

## Items already complete
1. Continue button disabled until ready (already completed in HS-1789)
1. Show password independently (done).
image
1. Editable command window (so they can copy and paste the right path) (done: Box was made into a textarea, and the styling was updated to match original design so it looks seamless. - ST)
image
1. Highlight the part that needs editing (done: This was a fairly complicated change because of the Download trigger directly before highlighting. We're entering the territory of the browser's security features (not allowing interaction/selection/highlighting with the page after a download until the user clicks on the page again)). This feature works to highlight the text to change when the user downloads, and then any time again when the user clicks on the textarea, and the text is still present in the string (in case they click away and then click again to try and see what needs to change)
image

1. HS-1807 - Added Tag to the First Discovery (one-line change)
1. HS-1806 - IPV6 removal (one-line change)
1. Docker Link Updated to mac/windows/linux (done)
image

1. Requirements expanded by default (done)
image

1. Timezones - Mike handled this in another ticket already, and DJ is working to remove the requirement to include them altogether.
1. Fixed a mobile bug where the dialogs/modals would not match the width of a small window.
1. Fixed a resize bug where the svg would not properly match the width and height of the window.
1. Update welcome text to provide more clear instruction (bonnie provided copy)

# Updates After Bonnie's Feedback:
image
image
image
image

### Welcome Screen

1. Update styling for requirements (requires visual direction)

### Minion Installation Screen
1. Update text to provide more information around new terms (requires new copy + visual direction)
1. Locations Box, we may want to consider letting users know the name of the location (requires visual direction)
1. Step 1: Fix File Name Path (requires new copy)
1. Might need a time lapsed timer on the page (requires visual direction)
1. Add text for the user to remind them to save the command line somewhere (requires visual direction + copy)

### Add Device Screen
1. Provide an error message for users who may input an ip range (requires copy for this situation)

## Jira link(s)
- https://opennms.atlassian.net/browse/HS-1802

## Checklist
* [X] Follows Lōkahi's [development guidelines.](https://github.com/OpenNMS-Cloud/lokahi/wiki/Development-Guidelines)
* [x] Appropriate reviewer(s) have been selected.
* [x] Jira issue(s) have been updated to "In Review".
* [x] Includes [appropriate tests.](https://github.com/OpenNMS-Cloud/lokahi/wiki/Test-Strategy)
* [x] Documentation has been updated as necessary.
* ~~[ ] Notify devops of changes to the Charts.~~
* ~~[ ] Notify documentation team of any changes to names of screens or features (affects URLs).~~

### CODE REVIEW

**Review by mikewrosey — APPROVED** [review]

File: ui/src/components/Welcome/WelcomeSlideOne.vue [hunk 1]
```diff
@@ -2,11 +2,12 @@
...
     ]">
         <div class="welcome-text">
-            <h1 data-test="welcome-slide-one-title">Welcome to <br />OpenNMS&nbsp;Cloud!</h1>
-            <p class="margin-bottom">Our mission is to make monitoring just&nbsp;happen.</p>
-            <p class="margin-bottom">To start monitoring, your network must include at least one OpenNMS Minion for
-                distributed network monitoring. We'll walk you through that setup, which takes only a
-                few&nbsp;minutes.</p>
+            <h1 data-test="welcome-slide-one-title">Welcome to <br />OpenNMS&nbsp;Cloud</h1>
+            <p class="margin-bottom">To start monitoring, you need to install our Minion on your system. This lightweight,
+                secure collector monitors and communicates with the devices on your network and shares collected data with
```
Comment (Bonrob2) [inline ui/src/components/Welcome/WelcomeSlideOne.vue:7]:
```suggestion
                secure collector monitors and communicates with the devices on your network, shares collected data,
```

File: ui/src/components/Welcome/WelcomeSlideOne.vue [hunk 2]
```diff
@@ -2,11 +2,12 @@
...
+                OpenNMS Cloud. The Minion connects your network with OpenNMS&nbsp;Cloud. </p>
```
Comment (Bonrob2) [inline ui/src/components/Welcome/WelcomeSlideOne.vue:8]:
```suggestion
and connects your network with OpenNMS&nbsp;Cloud. </p>
```
Tightened it up from the original.

File: ui/src/components/Welcome/WelcomeSlideOne.vue [hunk 3]
```diff
@@ -54,11 +55,24 @@
                     <li>Docker environment with Docker Compose</li>
                     <li>Network Time Protocol (NTP) installed and configured</li>
```
Comment (Bonrob2) [inline ui/src/components/Welcome/WelcomeSlideOne.vue:56]: Could we add the following so they know they can't do this on their phone?
```suggestion
                    <li> Access to a terminal window
                    <li>Network Time Protocol (NTP) installed and configured</li>
```

File: ui/src/components/Welcome/WelcomeSlideTwo.vue [hunk 4]
```diff
@@ -54,7 +54,7 @@
                 <div class="welcome-slide-step" data-test="welcome-page-two-internal">
                     <h2>Step 2: Copy and Run Docker Install Command</h2>
```
Comment (Bonrob2) [inline ui/src/components/Welcome/WelcomeSlideTwo.vue:55]:
```suggestion
                    <h2>Step 2: Copy and Run Docker Install Command in Terminal Window</h2>
```

File: ui/src/components/Welcome/WelcomeSlideTwo.vue [hunk 5]
```diff
@@ -54,7 +54,7 @@
...
                     <pre
-                        class="pre-wrap">Replace <strong>PATH_TO_DOWNLOADED_FILE</strong> with the full path to the certificate file.</pre>
+                        class="pre-wrap">Replace <strong>PATH_TO_DOWNLOADED_FILE</strong> with the full path to the certificate you downloaded. Remember to store your password, certificate and docker command securely. You need all three to run your minion.</pre>
```
Comment (Bonrob2) [inline ui/src/components/Welcome/WelcomeSlideTwo.vue:57]:
```suggestion
                        class="pre-wrap">In the command, replace <strong>PATH_TO_DOWNLOADED_FILE</strong> with the full path to the certificate you downloaded. Remember to store your password, certificate and Docker command securely. You need all three to run your Minion.</pre>
```

File: ui/src/components/Welcome/WelcomeSlideThree.vue [hunk 6]
```diff
@@ -10,7 +10,7 @@
                 <FeatherInput label="Name" data-test="welcome-slide-three-name-input"
                     :modelValue="welcomeStore.firstDiscovery.name" :error="welcomeStore.firstDiscoveryErrors.name"
                     @update:modelValue="(e) => welcomeStore.updateFirstDiscovery('name', e)" />
-                <FeatherInput label="Management IPV4/IPV6" data-test="welcome-slide-three-ip-input"
```
Comment (Bonrob2) [inline ui/src/components/Welcome/WelcomeSlideThree.vue:13]: In line 7 above, could you add "address" after IP? ("... associated with the specified IP address.")

File: ui/src/components/Welcome/WelcomeSlideTwo.vue (hunk 4 above)
Comment (Bonrob2) [reply ui/src/components/Welcome/WelcomeSlideTwo.vue:55]: Also, could you update lines 27 and 30 above?
27: Step 1: Download Encrypted Minion Certificate
30: Encrypted Minion Certificate

File: ui/src/store/Views/welcomeStore.ts [hunk 7]
```diff
@@ -299,6 +319,9 @@ export const useWelcomeStore = defineStore('welcomeStore', {
         this.minionStatusCopy = 'Minion detected.'
       }
```
Comment (Bonrob2) [inline ui/src/store/Views/welcomeStore.ts:320]: Could you add the following to line 316: "This can take up to 10 minutes."

**Review by Bonrob2 — CHANGES_REQUESTED** [review]
A few content changes. I believe there are some additional ones with regard to location (not included as feedback), but not covered under this task, so we can defer until Amanda has spoken with Jose.

File: ui/src/components/Welcome/WelcomeSlideOne.vue (hunk 3 above)
Comment (scthompsonopen) [reply ui/src/components/Welcome/WelcomeSlideOne.vue:57]: Done.

File: ui/src/components/Welcome/WelcomeSlideTwo.vue (hunk 4 above)
Comment (scthompsonopen) [reply ui/src/components/Welcome/WelcomeSlideTwo.vue]: All 3 should now be done.

File: ui/src/store/Views/welcomeStore.ts (hunk 7 above)
Comment (scthompsonopen) [reply ui/src/store/Views/welcomeStore.ts:320]: Done. I'm going to quickly generate a screenshot of what this looks like now, as I think the text wraps (still looks okay)

File: ui/src/components/Welcome/WelcomeSlideTwo.vue [hunk 8]
```diff
@@ -24,10 +24,10 @@
                 </div>
             </div>
             <div class="welcome-slide-step">
```
Comment (Bonrob2) [inline ui/src/components/Welcome/WelcomeSlideTwo.vue:26]: Sorry I missed this. Replace line 12 with:
The Minion installation can take up to 10 minutes. You will download an encrypted certificate for the Minion and run a Docker command.

The current text about virtual/hardware device is incorrect.

**Review by Bonrob2 — CHANGES_REQUESTED** [review]
Missed one update where the original is actually incorrect. Everything else looks great.

**Review by Bonrob2 — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**